projects/canopy/src/lib/alert/alert.component.ts

```diff
@@ -46,12 +47,22 @@ export class LgAlertComponent {
     this.variant = 'generic';
   }
 
+  ngOnChanges() {
+    this.initRole();
+  }
+
   @Input() set role(role: string) {
-    this._role = role;
+    this.explicitRole = role;
   }
-  private setRole() {
-    if (!this._role && this.variant !== 'info' && this.variant !== 'generic') {
-      this._role = 'alert';
+  private initRole() {
+    if (this.explicitRole) {
+      if (this.explicitRole !== 'none') {
```

Comment:
Personally I'd combine the two conditions into one here, but it's a matter of preference and no biggy

Reply:
If the conditions are combined, the logic will change.
Currently, if `explicitRole` exists, the `if` clause is fulfilled and it will not spill over to the next clauses.
If the condition would be for `explicitRole` to exist AND for the `explicitRole` not to be `none`, when `explicitRole` would be 'none' it would spill over to the next else clauses. As such, if it was for the `explicitRole` to be `none` and `variant` to be `warning` it would set the role to `alert`. 

I will refactor the code to be more clear tho.